src/main/java/DilemmaDetector/Consequences/DecisionCostCalculator.java

```diff
@@ -42,13 +56,22 @@ private int calculateCostOfHealthConsequence(Decision decision) {
         return result;
     }
 
-    // TODO sprawdzenie czy paseżer z głównego pojazdu
+    private boolean humanInsideMainVehicle(Vehicle mainVehicle, String victimName) {
+        if (mainVehicle == null)
+            return false;
+        return mainVehicle.getVehicle_has_passenger().stream().map(p -> ((Passenger) p).getOwlIndividual().getIRI().toString())
+                .anyMatch(p -> p.equals(victimName));
+    }
+
     private int calculateCostOfHealthConsequenceOfType(Decision decision, ConsequenceType consequenceType) {
         int result = 0;
         List<String> victims = consequenceContainer.getHealthConsequencesOfType(decision, consequenceType);
         for (String victimName : victims) {
             HashMap<PhilosophyParameter, Integer> parameters = customPhilosophy.getParameters();
-            if (factory.getHuman(victimName) != null) {
+            //extract scenario number to get main vehicle
+            String scenarioNumber = victimName.split("_")[0];
```

Comment:
Wyciągam main vehicle korzystając z tego że wszystkie obiekty mają ten sam numer

Reply:
Ogólnie to słabo trochę tak obchodzić ale działa

Reply:
Może zamiast tutaj wyciągać go, to przekazywać go przeze cały nasz proces? To można osobno zrobić, nie musi być w tym pr, ale trzeba zadanie na trello dać